jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1441,18 +1458,27 @@ ecma_builtin_typedarray_prototype_sort_compare_helper (ecma_value_t lhs, /**< le
 
   ecma_value_t compare_args[] = { lhs, rhs };
 
-  ECMA_TRY_CATCH (call_value,
-                  ecma_op_function_call (comparefn_obj_p,
-                                         ECMA_VALUE_UNDEFINED,
-                                         compare_args,
-                                         2),
-                  ret_value);
+  ecma_value_t call_value = ecma_op_function_call (comparefn_obj_p,
+                                                   ECMA_VALUE_UNDEFINED,
+                                                   compare_args,
+                                                   2);
+
+  if (ECMA_IS_VALUE_ERROR (call_value))
+  {
+    return call_value;
+  }
 
   if (!ecma_is_value_number (call_value))
   {
-    ECMA_OP_TO_NUMBER_TRY_CATCH (ret_num, call_value, ret_value);
+    ecma_number_t ret_num;
```

Comment:
I'd reverse the logic. If call value is number, return with it, otherwise try to convert it to number.